actions/setup-sigstore-env/action.yml

```diff
@@ -0,0 +1,47 @@
+# Copyright 2025 The Sigstore Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: setup-sigstore-env
+outputs:
+  oidc-url:
+    description: The url of the OIDC provider.
+    value: ${{ steps.run-containers.outputs.oidc-url }}
+  tsa-url:
+    description: The url of the Timestamp Authority (TSA).
+    value: ${{ steps.run-containers.outputs.tsa-url }}
+  trusted-root:
+    description: Path to the trusted root json file.
+    value: ${{ steps.run-containers.outputs.trusted-root }}
+  signing-config:
+    description: Path to the singning config json file.
+    value: ${{ steps.run-containers.outputs.signing-config }}
+runs:
+  using: composite
+  steps:
+    - uses: actions/setup-go@f111f3307d8850f501ac008e886eec1fd1932a34 # v5.3.0
+      with:
+        go-version: "stable"
+        cache-dependency-path: "**/go.sum"
+    - uses: ko-build/setup-ko@d982fec422852203cfb2053a8ec6ad302280d04d # v0.8
+    # Pending https://github.com/sigstore/cosign-installer/issues/187.
+    # - uses: sigstore/cosign-installer@d7d6bc7722e3daa8354c50bcb52f4837da5e9b6a # v3.8.1
+    #   with:
+    #     cosign-release: main
+    - shell: bash
+      run: go install github.com/sigstore/cosign/v2/cmd/cosign@main
```

Comment:
this cosign dependency is a little annoying for two reasons:
* it makes the local testing that bit more complicated (it's fun to run a 10 minute test process to find out I should have installed cosign first)
* this shadows any cosign install that the calling workflow already did and we probably don't want to break things like cosigns own tests like that 

some sort of container approach sounds better but maybe we leave that as followup?

Reply:
changed to run the cosign container.